provision/kubernetes/deploy.go

```diff
@@ -1413,6 +1502,11 @@ func (m *serviceManager) ensureServices(ctx context.Context, a provision.App, pr
 		if err != nil {
 			return errors.WithMessage(err, "could not to parse base services annotations")
 		}
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[backendConfigKey] = fmt.Sprintf("{\"default\":\"%s\"}", backendConfigName)
```

Comment:
This value cannot be a constant. It must be the name of the backend config created for this app/process. The name helper function I mentioned above can be useful here as well.